# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: 3a8153d6d830
# Full Hash: 3a8153d6d8304bef116f0f4dbedbabc0cc1a9ab7
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2023-01-24 17:01:02
# Regressor Bug: 1811195
# File Overlap Count: 1
# Description:
#   Bug 1811195 - Bail out on AppShutdownConfirmed in SharedWorkerManager::MaybeCreateRemoteWorker and BackgroundParentImpl::AllocPRemoteWorkerControllerParent r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167565
# ==============================================================================

diff -r df20721143f6 -r 3a8153d6d830 ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Tue Jan 24 11:22:52 2023 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Tue Jan 24 11:22:53 2023 +0000
@@ -544,6 +544,12 @@
 dom::PRemoteWorkerControllerParent*
 BackgroundParentImpl::AllocPRemoteWorkerControllerParent(
     const dom::RemoteWorkerData& aRemoteWorkerData) {
+  // Creating RemoteWorkerController may result in creating new processes,
+  // but during parent shutdown that would add just noise, so better bail out.
+  if (AppShutdown::IsInOrBeyond(ShutdownPhase::AppShutdownConfirmed)) {
+    return nullptr;
+  }
+
   RefPtr<dom::RemoteWorkerControllerParent> actor =
       new dom::RemoteWorkerControllerParent(aRemoteWorkerData);
   return actor.forget().take();
